File: workspace_java/BooleanExpression/src/br/unicamp/ic/bool/expression/BooleanNOT.java
package br.unicamp.ic.bool.expression;

import br.unicamp.ic.bool.BooleanStore;

public class BooleanNOT extends BooleanNode{
	private BooleanNode node;
	
	public BooleanNOT(BooleanNode points)
	{
		super(bool_type.NOT);
		this.node = points;
	}
	
	public BooleanNode getElem()
	{
		return node;
	}
	
	public boolean compute(BooleanStore bs)
	{
		return !node.compute(bs);
	}
	
	public String toString()
	{
		return "!" + node.toString();
	}
}

File: workspace_cpp/BooleanGraph/src/boolean/nodes/boolean_not.cpp
/*
 * BooleanNOT.cpp
 *
 *  Created on: 5 de set de 2015
 *      Author: vroth
 */

#include "boolean_not.hpp"

BooleanNOT::BooleanNOT(BooleanNode * down) : down(down) { btype = BOOLEAN_NOT; }

BooleanNOT::~BooleanNOT() { }

std::string BooleanNOT::toString() {
	return std::string("!" + down->toString());
}

bool BooleanNOT::compute(BooleanStore *bs) {
	return !down->compute(bs);
}

BooleanNode * BooleanNOT::getDown() {
	return down;
}
